client/ayon_core/plugins/publish/collect_farm_target.py

```diff
@@ -14,22 +14,20 @@ def process(self, instance):
         if not instance.data.get("farm"):
             return
 
-        context = instance.context
+        addons_manager = instance.context.data.get("ayonAddonsManager")
 
-        farm_name = ""
-        addons_manager = context.data.get("ayonAddonsManager")
-
-        for farm_renderer in ["deadline", "royalrender"]:
+        farm_renderer_addons = ["deadline", "royalrender"]
+        for farm_renderer in farm_renderer_addons:
             addon = addons_manager.get(farm_renderer, False)
-
-            if not addon:
-                self.log.error("Cannot find AYON addon '{0}'.".format(
-                    farm_renderer))
-            elif addon.enabled:
+            if addon.enabled:
```

Comment:
Wouldn't this fail if addon is not found?

Reply:
You are one hundred percent correct - fixed with https://github.com/ynput/ayon-core/pull/637/commits/3b5791a8e08c8b9831b2ad95a2511006fcf3478c